EdgeTagSDK/Networking/Manager/NetworkManager.swift

```diff
@@ -267,7 +268,129 @@ public class NetworkManager
+            completion(false,error)
+            print("\(error.rawValue)")
+            return
+        }
+        
+        for value in idGraph.values
+        {
+            if !(value is String || value is NSNumber || value is Bool || value is Int || value is UInt || value is Int8 || value is UInt8 || value is Int32 || value is UInt32 || value is Int64 || value is UInt64 || value is Double || value is Float)
+            {
+                let error = UserKeyError.invalidDataValueType
+                completion(false,error,nil)
+                print("\(error.rawValue)")
+                return
+            }
+        }
+        
+        let useragent = getUserAgent()
+        let storageDict = UserDefaults.standard.object(forKey: Constants.storageParameter) ?? [:]
+        let cookieHeader = StorageHandler.shared.getCookieForHeader()
+        let pageURL = PackageProviders.shared.getScreenName()
+        
+        router.request(.data(idGraph: idGraph, storage: storageDict as! Dictionary<AnyHashable, Any>, userAgent: useragent, cookieStr: cookieHeader, pageURL: pageURL)) { data, response, error in
+            
+            if let response = response as? HTTPURLResponse  {
+                let result = self.handleNetworkResponse(response)
+                switch result {
+                case .success:
+                    print("data id graph api success")
+                    completion(true,nil)
+                    break
+                case .failure(_):
+                    completion(false,error)
+                    break
+                }
+            }
+        }
+    }
+    
+    public func getDataForIDGraphKeys(idGraphKeys:Array<String>,completion: @escaping (_ success:Bool, _ error: Error?, _ idGraph:Dictionary<String,AnyHashable>?) -> Void)
+    {
+        if !isSDKInitialized
+        {
+            let error = UserKeyError.sdkUninitialized
+            completion(false,error,nil)
+            print("\(error.rawValue)")
+            return
+        }
+        
+        let cookieHeader = StorageHandler.shared.getCookieForHeader()
+        
+        router.request(.getData(dataKeys: idGraphKeys, cookieStr: cookieHeader)) { data, response, error in
+            
+            if let response = response as? HTTPURLResponse  {
+                let result = self.handleNetworkResponse(response)
+                switch result {
+                case .success:
+                    print("data id graph api success")
+                    if data != nil{
+                        do {
+                            let jsonDict =  try JSONSerialization.jsonObject(with: data!, options: []) as? [String: AnyHashable]
+                            completion(true,nil,jsonDict)
+                        } catch {
+                            print(error.localizedDescription)
+                        }
+                    }
+                    else
+                    {
+                        completion(true,nil,nil)
```

Comment:
Here , we must send empty hashmap. otherwise integration partner have to handle this case in different way.

Reply:
fixed